lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -126,32 +126,76 @@ def _proc_upload_reads_params(self, params):
+                          "include").format(",".join(parameters_should_unfilled)))
+                raise ValueError(("'source_reads_ref' was passed, making the following list of " +
+                                  "parameters : {} erroneous to " +
+                                  "include").format(", ".join(parameters_should_unfilled)))
+            # Check that it is a reads object. If not throw an eror.
+            single_input, kbasefile = self.check_reads(source_reads_object)
+            o = {'sequencing_tech': source_reads_object['data'].get("sequencing_tech"),
+                 'single_genome': source_reads_object['data'].get("single_genome")
+                 }
+            if 'strain' in source_reads_object['data']:
+                o['strain'] = source_reads_object['data']['strain']
+            if 'source' in source_reads_object['data']:
+                o['source'] = source_reads_object['data']['source']
+            if not single_input and not single_end:
+                # is a paired end input and trying to upload a filtered/trimmed
+                # paired end ReadsUtils. need to check for more fields.
+                ism = source_reads_object['data'].get('insert_size_mean')
+                issd = source_reads_object['data'].get('insert_size_std_dev')
+                read_orientation_out = source_reads_object['data'].get('read_orientation_outward')
+                o.update({'insert_size_mean': ism,
+                          'insert_size_std_dev': issd,
+                          'interleaved': interleaved,
+                          'read_orientation_outward': read_orientation_out
+                          })
+        else:
+            seqtype = params.get('sequencing_tech')
+            if not seqtype:
+                raise ValueError('The sequencing technology must be provided')
+            sg = 1
+            if 'single_genome' in params and not params['single_genome']:
+                sg = 0
+            o = {'sequencing_tech': seqtype,
+                 'single_genome': sg
+                 # ,
+                 # 'read_count': params.get('read_count'),
+                 # 'read_size': params.get('read_size'),
+                 # 'gc_content': params.get('gc_content')
+                 }
+            self._add_field(o, params, 'strain')
+            self._add_field(o, params, 'source')
+            ism = params.get('insert_size_mean')
+            self._check_pos(ism, 'insert_size_mean')
+            issd = params.get('insert_size_std_dev')
+            self._check_pos(issd, 'insert_size_std_dev')
+            if not single_end:
+                read_orientation_out = 1 if params.get('read_orientation_outward') else 0
+                o.update({'insert_size_mean': ism,
+                          'insert_size_std_dev': issd,
+                          'interleaved': interleaved,
+                          'read_orientation_outward': read_orientation_out
+                          })
```

Comment:
This function is now way too long - needs to be split up between multiple sub functions.